### PR TITLE
Create NodeState to group parameters to consensus state

### PR DESCRIPTION
**Pull request by ariqchowdhury**
This change groups common state parameters into consensus state handlers. This also simplifies consensus state unit tests

### CODE REVIEW

File: monad-consensus-state/src/lib.rs [hunk 1]
```diff
@@ -176,6 +176,22 @@ pub enum StateRootAction {
     Defer,
 }
 
+pub struct NodeState<'a, ST, SCT, VTF, LT, TT>
```
Comment (michael-yxchen) [inline monad-consensus-state/src/lib.rs:179]: nit: monad-node also defines a [NodeState](https://github.com/monad-crypto/monad-bft/blob/f9c4204e53396c1c0afba7a883a9eef032241cee/monad-node/src/state.rs#L18). Though we won't be using the consensus NodeState outside of monad-state, renaming this one can help avoid ambiguity. One idea I have is `ConsensusContext`. 

I don't feel strong about it. Feel free to leave as is or pick other names.

File: monad-consensus-state/src/lib.rs [hunk 2]
```diff
@@ -1353,20 +1553,57 @@ mod test {
...
+        let (env, mut ctx) =
+            setup::<SignatureType, SignatureCollectionType, _, StateRootValidatorType, _, _>(
+                num_state as u32,
+                ValidatorSetFactory::default(),
+                SimpleRoundRobin::default(),
+                MockTxPool::default(),
+                || NopStateRoot,
+            );
+
+        let (consensus_state, mut node_state) = ctx[0].get_state();
```
Comment (michael-yxchen) [inline monad-consensus-state/src/lib.rs:1605]: I see that some tests splits `NodeContext` to `ConsensusState` and others call the utility function from `NodeContext`. Just for my understanding, how does a test choose one way or the other?

File: monad-consensus-state/src/lib.rs (hunk 1 above)
Comment (ariqchowdhury) [reply monad-consensus-state/src/lib.rs:179]: I'll leave it for now, if it ends up being annoying can rename later

File: monad-consensus-state/src/lib.rs (hunk 2 above)
Comment (ariqchowdhury) [reply monad-consensus-state/src/lib.rs:1605]: probably related to borrow checking

**Pull request merged**